spec/components/attachment_link_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+describe "Attachment Link", type: :view do
+  def component_name
+    "attachment_link"
+  end
+
+  it "fails to render when no attachment is given" do
+    assert_raises do
+      render_component({})
+    end
+  end
+
+  it "renders an attachment link" do
+    render_component(attachment: { title: "Attachment", url: "https://gov.uk/attachment" })
+    assert_select ".gem-c-attachment-link"
+    assert_select 'a[href="https://gov.uk/attachment"]', text: "Attachment"
+  end
+
+  it "can have a target specified" do
+    render_component(
+      attachment: { title: "Attachment", url: "attachment" },
+      target: "_blank",
+    )
+    assert_select 'a[href=attachment][target=_blank]'
+  end
+
+  it "can include attribute metadata" do
+    render_component(
+      attachment: {
+        title: "Attachment",
+        url: "attachment",
+        content_type: "text/xml",
+        file_size: 2048,
+        number_of_pages: 2,
```

Comment:
Should there be a test that 1 page outputs the text `page` as opposed to `pages`?

Reply:
I don't think that's necessary as that is testing pluralize that ActionView provides. If we were to test that it'd beg the question as to whether to test all the content_type inputs and file_size variations.